### PR TITLE
[WIP] Implement #lolcode

### PR DESCRIPTION
**Pull request by CodaFi**
This is the corresponding implementation of [#lolcode](https://github.com/apple/swift-evolution/pull/823).  

We are still working through the particulars of loops and switches because [the spec](https://github.com/justinmeza/lolcode-spec/blob/master/v1.2/lolcode-spec-v1.2.md) is quite vague on some points, hence WIP.

Also, while implementing this we realized that LOLCode reserves A, Z, O, and R as keywords - and stdlib uses those as generic argument names.  It also reserves `NO`, so I took the opportunity to just remove that little error we make for ObjC `NO`.  YAGNI, or something.

### CODE REVIEW

**Comment by CodaFi** [comment]
Let's see what horrors await

@swift-ci please smoke test

**Comment by harlanhaskins** [comment]
@codafi and I are thrilled to bring forward this groundbreaking feature on such an important day.

File: include/swift/AST/DiagnosticsParse.def [hunk 1]
```diff
@@ -1245,6 +1245,22 @@ ERROR(expr_selector_expected_property_expr,PointsToFirstBadToken,
...
+ERROR(expr_lolcode_expected_lparen,PointsToFirstBadToken,
+      "expected '(' following '#lolcode'", ())
+ERROR(expr_lolcode_expected_rparen,PointsToFirstBadToken,
+      "expected ')' to complete '#lolcode' expression", ())
+ERROR(statement_lolcode_same_line_without_comma,none,
+      "consecutive statements on a line must be separated by ','", ())
+ERROR(statement_lolcode_no_yarly,none,
+      "expected 'YA RLY' statement after 'O RLY?' statement", ())
+ERROR(statement_lolcode_aggressive_orly,none,
+      "'O RLY?' statement must be phrased as a question", ())
```
Comment (jrose-apple) [inline include/swift/AST/DiagnosticsParse.def]: It seems unfortunate that the error messages for LOLCODE ([sic](http://lolcode.org)) are in the same formal tone as the ones for the rest of the compiler. To make this part of the language more compelling, I suggest having form follow…something…and taking inspiration from the syntax itself:

```
ERROR(expected_of_lolcode_builtin,PointsToFirstBadToken,
      "ABSENCE OF 'OF" -> NO COMPILER LOVE", ())

ERROR(statement_lolcode_same_line_without_comma,none,
      "WHY CANT I HOLD ALL THESE STATMENTS", ())
ERROR(statement_lolcode_no_yarly,none,
      "DONT LEAV ME HANGIN", ())
ERROR(statement_lolcode_aggressive_orly,none,
      "Q U E S T I O N   E V E R Y T H I N G", ())
```

**Comment by harlanhaskins** [comment]
This...might also be the first LOLCode to LLVM compiler in existence...

**Pull request closed without merging**